src/ReachSets/discretize.jl

```diff
@@ -647,6 +682,45 @@ function discretize_interpolation(𝑆::InitialValueProblem{<:AbstractContinuous
     else
         throw(ArgumentError("input of type $(typeof(U)) is not allowed"))
     end
+    return Ω0, Ud
+end
 
-    return IVP(CLCDS(ϕ, Id(size(A, 1)), nothing, Ud), Ω0)
+# version using concrete operations with zonotopes
+function _discretize_interpolation_homog(X0, ϕ, Einit, set_operations::Val{:zonotope})
+    Einit = convert(Zonotope, Einit)
```

Comment:
For type stability I would use a new variable (also in the other function).
Ideally, copy this method with a type annotation `::Zonotope` and from the old method call the new method with the `convert`:
```julia
function _discretize_interpolation_homog(X0, ϕ, Einit, set_operations::Val{:zonotope})
    return _discretize_interpolation_homog(X0, ϕ, convert(Zonotope, Einit), set_operations)
end

function _discretize_interpolation_homog(X0, ϕ, Einit::Zonotope, set_operations::Val{:zonotope})
    ...
end
```

Reply:
Since this function is used internally i preferred to not annotate the types here -- at compile time Julia will find the correct function arguments at compile time, without instability:

```julia
julia> using Revise, Test, LazySets

julia> function f(x) x + [1.0] end
f (generic function with 1 method)

julia> Test.@inferred f([1.0])
1-element Array{Float64,1}:
 2.0

julia> Test.@inferred f(Singleton([1.0]))
Translation{Float64,Array{Float64,1},Singleton{Float64,Array{Float64,1}}}(Singleton{Float64,Array{Float64,1}}([1.0]), [1.0])

```

Reply:
My original point was that `Einit` changes its type. Using two methods is one solution, using a new variable is another solution.

Reply:
Ok, i see. Not only `Einit` are converted to Zonotope, also `X0`, `Eψ0`, `U0`, `Uk` may change its type. I think that we are good with these conversions, they are done only once.

Reply:
Yes, this code is not performance critical. But in general [this should be avoided](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-changing-the-type-of-a-variable-1).

Reply:
Yes. The conversions do not happen in a loop in this function. Shall we mark this issue are resolved?

Reply:
I think this is more general than loops, but yes.